http/server/server.go

```diff
@@ -281,13 +285,38 @@ func (s HTTPService) History(w http.ResponseWriter, r *http.Request) {
 
 func (s HTTPService) GetConfig(w http.ResponseWriter, r *http.Request) {
 	inst := getInstanceID(r)
-	config, err := s.service.GetConfig(inst)
+	fingerprint := r.FormValue("fingerprint")
+	config, err := s.service.GetConfig(inst, fingerprint)
 	if err != nil {
 		errorResponse(w, r, err)
 		return
 	}
 
-	jsonResponse(w, r, config)
+	safeConfig := config.HideSecrets()
```

Comment:
Worth a comment saying why this is necessary (I immediately thought: "hang on, it didn't used to do this!")